### PR TITLE
Imrove metrics collection

### PR DESCRIPTION
**Pull request by foreverjun**
- Collect metrics to csv

### CODE REVIEW

**Comment by foreverjun** [comment]
Добавил графики с lowess аппроксимацией и доверительными интервалами. Иногда это работает нормально а иногда нет. Математику доразберу чуть позже.
image
image

File: src/serializer.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+use serde::ser::SerializeStruct;
+use tokio_metrics::RuntimeMetrics;
+
+#[derive(serde::Serialize)]
+pub struct MetricsSerializable{
```
Comment (IgorErin) [inline src/serializer.rs:5]: Metrics already serializable. Take a look at https://github.com/IgorErin/tokio-metrics/tree/iea/submodule

File: Cargo.toml [hunk 2]
```diff
@@ -10,6 +10,7 @@ itertools = "0.13.0"
 tokio-metrics = { path = "tokio-metrics", features = ["rt"] }
 serde = "1.0.215"
 serde_json = "1.0.132"
+csv = "1.3.1"
```
Comment (IgorErin) [inline Cargo.toml:13]: Why not json? Can we use serde with cvs if we want?

**Review by IgorErin — COMMENTED** [review]
Can we reuse tokio-metrics submodule features?

File: src/serializer.rs (hunk 1 above)
Comment (foreverjun) [reply src/serializer.rs:5]: There Duration for several metrics is serialized into 2 values, and I just represent it in microseconds. Also I want to add iteration number and not collect unnecessary metrics(poll_time_histogram). Nested structures are not serialized in csv

**Pull request closed without merging**